crates/llm-local/src/lib.rs

```diff
@@ -32,6 +33,7 @@ pub struct LocalLlmEngine {
 
 #[async_trait]
 impl LlmEngine for LocalLlmEngine {
+    #[instrument(skip(self, prompt))]
     async fn infer(
```

Comment:
I forget - by default what is the name of the span generated here? Do we want the spans for this to just say "infer"? Would be nice if it were a bit more clear from span name alone exactly what was happening. Something like "llm_generate_local_inference" and "llm_generate_local_embeddings".

Reply:
Sounds good - as I've previously said, it would be amazing if we had some sort of system for deciding how to name spans. As we add more and more of these, we're going to simply be relying on author/reviewer judgement as to what a good span name is - it would be nicer if we gave some sort of guidance.

Reply:
Names are updated.

To close the loop on the naming stuff: I'll flesh out the requirements for span names in my SIP.